File: app/src/main/java/nl/acidcats/tumblrlikes/LikesApplication.java
package nl.acidcats.tumblrlikes;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.facebook.stetho.Stetho;
import com.google.firebase.analytics.FirebaseAnalytics;

import io.fabric.sdk.android.Fabric;
import nl.acidcats.tumblrlikes.di.AppComponent;
import nl.acidcats.tumblrlikes.di.AppModule;
import nl.acidcats.tumblrlikes.di.DaggerAppComponent;
import nl.acidcats.tumblrlikes.di.DataModule;
import timber.log.Timber;

/**
 * Created by stephan on 28/03/2017.
 */

public class LikesApplication extends Application {
    private static final String TAG = LikesApplication.class.getSimpleName();

    private AppComponent _appComponent;
    private boolean _isFreshRun = true;

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());

        if (BuildConfig.DEBUG) {
            Stetho.initializeWithDefaults(this);
        }

        FirebaseAnalytics analytics = FirebaseAnalytics.getInstance(this);

        _appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this, analytics))
                .dataModule(new DataModule(this))
                .build();

        Timber.plant(new Timber.DebugTree());

    }

    public AppComponent getAppComponent() {
        return _appComponent;
    }

    public boolean isFreshRun() {
        return _isFreshRun;
    }

    public void setFreshRun(boolean freshRun) {
        _isFreshRun = freshRun;
    }
}

File: app/src/main/java/nl/acidcats/tumblrlikes/LikesApplication.kt
package nl.acidcats.tumblrlikes

import android.app.Activity
import android.app.Application
import android.app.Service
import com.crashlytics.android.Crashlytics
import com.facebook.stetho.Stetho
import com.github.ajalt.timberkt.Timber
import com.github.ajalt.timberkt.Timber.DebugTree
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import dagger.android.HasServiceInjector
import io.fabric.sdk.android.Fabric
import nl.acidcats.tumblrlikes.di.AppInjector
import javax.inject.Inject

/**
 * Created on 05/11/2018.
 */
class LikesApplication : Application(), HasActivityInjector, HasServiceInjector {

    @Inject
    lateinit var activityInjector: DispatchingAndroidInjector<Activity>
    @Inject
    lateinit var serviceInjector: DispatchingAndroidInjector<Service>

    var isFreshRun = true
        private set

    override fun onCreate() {
        super.onCreate()

        initDagger()
        initTimber()
        initFabric()
        initStetho()
    }

    private fun initDagger() {
        AppInjector.init(this)
    }

    private fun initStetho() {
        if (BuildConfig.DEBUG && BuildConfig.USE_STETHO) {
            Stetho.initializeWithDefaults(this)
        }
    }

    private fun initFabric() {
//        Fabric.with(this, Crashlytics())
    }

    private fun initTimber() {
        Timber.plant(DebugTree())
    }

    fun clearFreshRun() {
        isFreshRun = false
    }

    override fun activityInjector(): AndroidInjector<Activity> = activityInjector

    override fun serviceInjector(): AndroidInjector<Service> = serviceInjector
}
